readthedocs/search/management/commands/reindex_elasticsearch.py

```diff
@@ -0,0 +1,95 @@
+        chunk_objects = chunks(instance_ids, settings.ES_TASK_CHUNK_SIZE)
+
+        for chunk in chunk_objects:
+            data = {
+                'app_label': app_label,
+                'model_name': model_name,
+                'document_class': document_class,
+                'index_name': index_name,
+                'objects_id': chunk
+            }
+            yield index_objects_to_es_task.si(**data)
+
+    def _run_reindex_tasks(self, models):
+        for doc in registry.get_documents(models):
+            qs = doc().get_queryset()
+            instance_ids = list(qs.values_list('id', flat=True))
+
+            app_label = qs.model._meta.app_label
+            model_name = qs.model.__name__
+
+            index_name = doc._doc_type.index
+            timestamp = datetime.datetime.now().strftime('%Y%m%d%H%M%S')
+            new_index_name = "{}_{}".format(index_name, timestamp)
+
+            pre_index_task = create_new_es_index_task.si(app_label=app_label,
+                                                         model_name=model_name,
+                                                         index_name=index_name,
+                                                         new_index_name=new_index_name)
+
+            indexing_tasks = self._get_indexing_tasks(app_label=app_label, model_name=model_name,
+                                                      instance_ids=instance_ids,
+                                                      document_class=str(doc),
+                                                      index_name=new_index_name)
+
+            post_index_task = switch_es_index_task.si(app_label=app_label, model_name=model_name,
+                                                      index_name=index_name,
+                                                      new_index_name=new_index_name)
+
+            # Task to run in order to add the objects
+            # that has been inserted into database while indexing_tasks was running
+            missed_index_task = index_missing_objects_task.si(app_label=app_label,
+                                                              model_name=model_name,
+                                                              document_class=str(doc),
+                                                              indexed_instance_ids=instance_ids)
+
+            # http://celery.readthedocs.io/en/latest/userguide/canvas.html#chords
+            chord_tasks = chord(header=indexing_tasks, body=post_index_task)
+            # http://celery.readthedocs.io/en/latest/userguide/canvas.html#chain
+            chain(pre_index_task, chord_tasks, missed_index_task).apply_async()
+
+            message = ("Successfully issued tasks for {}.{}, total {} items"
+                       .format(app_label, model_name, len(instance_ids)))
+            log.info(message)
+
+    @staticmethod
+    def _get_models(args):
+        for model_name in args:
+            yield apps.get_model(model_name)
```

Comment:
This doesn't need to be broken out into it's own function. It just increases complexity for little value.